### ISSUE TITLE
ec2 describe-snapshots does not respect region flags

### ISSUE THREAD

**Issue by lockan** [aws/aws-cli#3808]
I'm trying to return a list of EBS snapshots for only a specific region. I've tried this a number of ways, but the region flags never seem to be respected, so that I get a list of all snapshots across all regions. 

e.g. 
I have >5000 snapshots on ap-southeast-2, but only 10 on us-west-2.  When I run the following:
`aws --profile myprofile --region us-west-2 describe-snapshots`
I get results for all 5000 + 10 results, when I would expect only the 10 from us-west-2. 

If I edit my `~/.aws/config` and set the region for `myprofile` to `region=us-west-2` explicitly, I still get snapshot results for all regions. The same happens if I set `AWS_DEFAULT_REGION` in the shell environment. 

No default region is specified in my `~/.aws/credentials` file.

**Comment by JordonPhillips** [comment]
I can't duplicate. Here's what I'm doing:

```
$ export ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
$ aws ec2 describe-snapshots --owner-ids $ACCOUNT_ID --query 'length(Snapshots)' --region us-west-2
1
$ aws ec2 describe-snapshots --owner-ids $ACCOUNT_ID --query 'length(Snapshots)' --region us-east-1
0
```

My only hunch is that you aren't setting `--owner-ids`, which would result in you seeing every publicly available snapshot (which is a lot). If that's not the case, you might check what endpoint the cli is connecting do. You can do this by appending `--debug` and then searching for `https` in the output, like so:

```
$ aws ec2 describe-snapshots --owner-ids $ACCOUNT_ID --region us-west-2 --page-size 1 --max-items 1 --debug 2>&1 | grep https
[...]
2018-12-19 16:53:12,321 - MainThread - urllib3.connectionpool - DEBUG - https://ec2.us-west-2.amazonaws.com:443 "POST / HTTP/1.1" 400 None
```

**Comment by lockan** [comment]
I tried adding `--owner-ids` as suggested, and that does result in only the snapshots from that account and region. However, I'm not clear on why I should need to add `--owner-ids` in order to filter the results by region. While you've provided a solution that does get me the results I'm looking for, it feels like a work-around since `--region` and `--owner-ids` are two contextually unrelated flags that I would expect to be mutually exclusive. As such, it's a usability issue where the behaviour doesn't match user expectations. 

If the default behaviour _without_ `--owner-ids` is to list all public snapshots for which I have create volume permissions as the help text suggests, I would still expect the `--region` flag to restrict those results to the region I've specified, the same as the cli would for any other region-based resource query.
